src/Nethermind/Ethereum.Test.Base/TestBlockhashProvider.cs

```diff
@@ -26,6 +26,8 @@ public class TestBlockhashProvider : IBlockhashProvider
     {
         public Keccak GetBlockhash(BlockHeader currentBlock, in long number)
         {
+            if (number != 0)
```

Comment:
shouldn't be number == 0 here?

Reply:
No, it is okay. From the issue: "Others return keccak(blocknumber) only for "available blocks" which means it returns keccak(blocknumber) only for blocknumber 0, all others are returned as 0" and that was confirmed in tests.